core-bundle/src/Image/Studio/Figure.php

```diff
@@ -120,8 +120,8 @@ public function getMetadata(): Metadata
             throw new \LogicException('This result container does not include metadata.');
         }
 
-        // Safely return as Closure will be evaluated at this point
-        return $this->metadata;
+        /** @var Metadata */
+        return $this->metadata; // the closure will be evaluated at this point
```

Comment:
What about using `@param-out` on the `resolveIfClosure` method? I'm not sure this can this be templated? 
see https://psalm.dev/docs/annotating_code/supported_annotations/#param-out

I'm thinking of sth like this: 
```
@template T
@psalm-param T|\Closure|null $property
@param-out T|null $property
```
----

If this does not work we maybe want to annotate L135/198 as well? (`array|null`)